parsl/executors/status_handling.py

```diff
@@ -17,7 +17,7 @@ def __init__(self, provider):
         self._provider = provider
         self._executor_bad_state = threading.Event()
         self._executor_exception = None
-        self._tasks = {}  # type: Dict[str, Future]
+        self._tasks = {}  # type: Dict[object, Future]
```

Comment:
this same change needs to happen on NoStatusHandlingExecutor

Reply:
No Ben, it can't, because self._tasks in NoStatusHandlingExecutor has no type annotation.

That said, I added that type annotation, which hopefully addresses what must have been your actual objection.

Reply:
This came from me validating this PR against my more heavily type-annotated branch - that was more aggressive about the _tasks annotation.